### PR TITLE
Switch to correct Xdebug port 9003

### PR DESCRIPTION
**Pull request by mgenereu**
Xdebug was hard to use with the existing setup.  This switches a few configuration items to make it easier to use the debugger.

- Port 80 and 8080 was causing confusion especially when auto-forwarding the ports and opening browsers automatically.
- Port 9003 was offering to auto-forward.  This never needs to forward as the VS Code debugger runs on localhost in the container.
- XDebug is on 9003 with the new version.  Using 9000 is just begging for confusion.
- Apache is always complaining about the ServerName so localhost is set as the default.
- The Apache VirtualHost is set to the same port as the main configuration.  This not matching caused side effects.

Also, regarding #1052, I use this setup with the VS Code debugger running whenever I call PHP directly or via Apache so I don't get the warning.

### CODE REVIEW

**Comment by Chuxel** [comment]
Hmmm - Re: switching to 9003, the challenge here is that updating the image can affect people that already have things set up for port 9000. So, it's a bit of a breaking change. Everything else seems like it would be a clean and a good update tho!

**Comment by mgenereu** [comment]
> it's a bit of a breaking change.

Is there a way to announce breaking changes?  People were already having trouble using it.  Is there a place to put documentation and announcements for the specific container? I would add additional documentation on how to use this container for PHP if I knew where to put it.

**Comment by mgenereu** [comment]
I can't figure out why the smoke test is failing.  I looked for hidden characters with `od` but it looks like healthy JSON.

**Comment by mgenereu** [comment]
Speaking of breaking changes, I forgot about this line that the upstream container recommends for development.  That's another big one that solved PHP problems downstream.
```
cp "$PHP_INI_DIR/php.ini-development" "$PHP_INI_DIR/php.ini"
```

**Comment by mgenereu** [comment]
I just made a new `launch.json` file and it automatically populated with port 9003.  This might not be that breaking per #695? Thoughts, @Chuxel?  With a little feedback, I can continue to refine.

File: containers/php/.devcontainer/devcontainer.json [hunk 1]
```diff
@@ -2,33 +2,37 @@
...
 	],
-
-	// Use 'forwardPorts' to make a list of ports inside the container available locally.
-	"forwardPorts": [8080],
-
+	// Use 'portsAttributes' to control auto forwarding of ports
+	"portsAttributes": {
+		"8080": {
+			"label": "Application",
+			"onAutoForward": "notify",
```
Comment (Chuxel) [inline containers/php/.devcontainer/devcontainer.json:27]: There is a hanging comma here that is causing the CI to fail.

**Review by Chuxel — COMMENTED** [review]
> I just made a new launch.json file and it automatically populated with port 9003. This might not be that breaking per https://github.com/microsoft/vscode-dev-containers/issues/695? Thoughts, @Chuxel? With a little feedback, I can continue to refine.

The problem here is that the image (mcr.microsoft.com/vscode/devcontainers/php) can be used independant of updates to the definition.  Any changes made to the image would affect existing users that have a .devcontainer folder in their project.  What #695 was about was actually changing what was in the image to work - which is why that was clean.

So the question becomes - Is a port update worth that problem?  We'd basically have to add some text in the terminal telling people to make changes since they otherwise would not know. We have a mechanism to do it, but it will drive issues and questions.

For port `80` for example, we could actually add `onAutoForward` with a value of `ignore`  in devcontainer.json, which avoids that problem. Within the container, port 80 along with 8080 would then still be available which wouldn't break anything existing.

One possible way to manage `9003` is to have a separate build argument that indicates what should happen that is used in both the `base.Dockerfile` (used to generate the image) and `Dockerfile` that could be used to set alternate options. If these are not set, then the config is as it is now, but if it is set, then it would take whatever the build arguments were set to.

> Speaking of breaking changes, I forgot about this line that the upstream container recommends for development. That's another big one that solved PHP problems downstream.
>
> > cp "$PHP_INI_DIR/php.ini-development" "$PHP_INI_DIR/php.ini"

Yeah it sounds like this is mainly log output differences given the description which shouldn't break people I would expect. What problems did you see it resolve down stream? Assuming its not removing functionality vereses the default config, that's a safer update IMHO.

Also +@joshspicer since PHP config in the default Codespaces image is the same as this.

File: containers/php/.devcontainer/devcontainer.json (hunk 1 above)
Comment (mgenereu) [reply containers/php/.devcontainer/devcontainer.json:27]: Didn't know `JSON with Comments` was a relaxed mode for parsing JSON.  That was a new one for me.

**Comment by mgenereu** [comment]
I get the existing developer and downstream breakage but as a new developer using the current extensions, things were broken and I had to do some digging.  I'll close this PR here to document for others what fixes made the developer experience better for me.

**Pull request closed without merging**